dotty_dict/dotty_dict.py

```diff
@@ -114,6 +114,32 @@ def search_in(items, data):
 
         return search_in(self._split(item), self._data)
 
+    @staticmethod
+    def _find_data_type(item, data):
+        """This method returns item in datatype that exists in data dict.
+
+        Method creates set of types present in dict keys
+        and then iterates through them trying to convert item
+        into one of types and check whether item under this type
+        exists in dict keys. If yes then it'll return converted item.
+        Otherwise item stays the same type as it was on entry.
+
+        :param item: Item to convert to proper type
+        :type item: any type
+
+        :return: Converted or unchanged item
+        :rtype: any type
+        """
+        data_types = set([type(i) for i in data.keys()])
+        for t in data_types:
+            try:
+                if t(item) in data:
+                    item = t(item)
+                    return item
+            except ValueError:
+                pass
+        return item
```

Comment:
What would happen if item will be None? Is that possible?

Reply:
Then it'll return None.

Reply:
I'll add unittest for this